src/Illuminate/Validation/Validator.php

```diff
@@ -747,7 +747,7 @@ protected function validateDifferent($attribute, $value, $parameters)
 
         $other = Arr::get($this->data, $parameters[0]);
 
-        return isset($other) && $value != $other;
+        return isset($other) && $value !== $other;
```

Comment:
I agree we probably can't change this right now, as its a breaking change and would cause things like "0" against 0 to fail.